forms/app/covid_trace_request.xml

```diff
@@ -0,0 +1,573 @@
+            <value>Contact information</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/address:label">
+            <value>Address</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/age_in_years:label">
+            <value>Age in years</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/chief_name:label">
+            <value>Name of the chief of the village</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/chief_phone:label">
+            <value>Phone of the chief</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/church:label">
+            <value>Church name</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/flight_details:label">
+            <value>Flight details (if relevant)</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/n_contact_location:label">
+            <value>Where the contact took place</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/other_names:label">
+            <value>Other names</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/parent/_id:label">
+            <value>Parent</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/parent/parent/_id:label">
+            <value>Area</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/parent/parent/parent/_id:label">
+            <value>Health Facility ID</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/phone:label">
+            <value>Phone No</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/sex/female:label">
+            <value>Female</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/sex/male:label">
+            <value>Male</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/sex:label">
+            <value>Sex</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/surname:label">
+            <value>Surname</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/town:label">
+            <value>Town</value>
+          </text>
+          <text id="/covid_trace_request/contacts/contact_repeat/contact_profile/tracer_area:label">
+            <value>Tracer Area</value>
```

Comment:
When i assign a tracer area ( for an existing tracer user ) , i should see the patient in the list of contacts when i connect as the tracer, but right now i don't. Also it seems like the patient is not being replicated in the database, i can see him in contact list of the data entry but not elsewhere

Reply:
I had mistakenly kept the patient zero at the lowest hierarchy and tested it from there. However, now I realize that the replication fails if the patient zero is kept at a higher level, as required by the workflow. I am sorry about that.
This is because the contacts created through the report have certain hierarchy information (depending on the contact's hierarchy level) and the server does not accept if the hierarchy is misaligned. I have tested a fix for it successfully.

After looking at the workflow again, I get the following understanding. Please let me know if any of it needs to be corrected:
1. A Data Entry user is always on Hierarchy 0.
2. A Tracer user is always on Hierarchy 1.
3. A Patient Zero contact is always on Hierarchy 0.
4. There are only two hierarchy levels (DE and Tracer Area).

Reply:
so ideally the data entry user will be the primary user attached to a `health_center` places and the trace will be the primary user attached to a `clinic` place. Contacts that needs to be traced should be members under the tracer's clinic. And patient zéro could be whether 
1. created as (not primary) contacts under the data entry place,
2. we could also create a special household/clinic that is not for any tracer but that is for all the patient Zeros (actually i think this would be ideal

Is any between 1. & 2. possible following your findings, reading your comment i feel like 2. is what you tested is it right ? If yes then it's okay, i tested with 1. but didn't expect it would fail.

Reply:
Also we should not forget to limit the access of the form in at the levels where it won't work.

Reply:
Scenario 1 is possible and is the default behavior. 2 might be a little harder but we can give a try.
I had first tested with a level 3 user and configured the hierarchy as such, but that can be fixed.

Is there still a `district_hosptial` place at the top level?

Reply:
I tested with 2. yesterday and that actually worked ( since you said the patient had to be at the lowest hierarchy, is that what you call level 3 ?  ) . But 1. didin't work when i first tried. So if 2. is working we can keep it like this it's the preferred behaviour. 

There is a district_hospital but it's not supposed to be used for this package

Reply:
Yes, for me:
Level 1: district-hospital (not used)
Level 2: health-center (Data Entry)
Level 3: clinic (Tracer Area)

To restrict access to the people/reports in level 3 for the data entry user (level 2), I was planning to use `"replication_depth": 1`.
If we do so and create a patient zero at level 3, won't that be a problem if the user needs to sign in from a different device?

Reply:
For now, Patient zero is created at the same level as the data entry user.
[Changes](https://github.com/medic/config-covid/pull/41/commits/ef70bd29ede65b8244f67dc3daa3ca1bcd357a5b):
Replication depth is set to 1 for `covid_data_entry` users.
Trace request action is only available to users associated with contact having role = "data_entry" and for contacts with role = "patient_zero"
Trace followup action is only available to users associated with contact having role = "tracer" and for contacts with role = "contact_n" (added automatically)
Trace followup task is only shown to users associated with contact having role = 'tracer'

Reply:
The replication depth is reverted to 3.
Trace request action is only available to the users associated with contact's role = "data_entry. It can be sent for the contacts at level 3 with role = "patient_zero"